### PR TITLE
feat(retry): added retry with max attempts

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**

### CODE REVIEW

File: confd/production.tmpl [hunk 1]
```diff
@@ -3,8 +3,8 @@
         "host_ip": {{ getv "/kafka/host/ip" "[\"10.28.11.49:9092\"]" }},
         "topic": "{{ getv "/kafka/topic" "topic-test-101" }}",
         "group_id": "{{ getv "/kafka/group/id" "task_group_1" }}",
-        "offset_reset": "{{ getv "/kafka/offest/reset" "latest" }}",
-        "poll_timeout_milliseconds": {{ getv "/kafka/poll/timeout/milliseconds" "270000000" }},
+        "offset_reset": "{{ getv "/kafka/offest/reset" "earliest" }}",
+        "poll_timeout_milliseconds": {{ getv "/kafka/poll/timeout/milliseconds" "8000" }},
```
Comment (asafMasa) [inline confd/production.tmpl]: revert

File: src/helper.py [hunk 2]
```diff
@@ -54,6 +56,17 @@ def save_update(self, taskId, status, fileName, progress=None, fullPath=None , d
...
             self.logger.error(f'Task Id "{taskId}" Failed to update database: {e}')
 
+    def get_status(self,taskId):
+        try:
+            self.logger.info(f'getting attmepts count for task "{taskId}"')
+            url = f'{self.url}/statuses/{taskId}'
+            res = requests.get(url=url)
+            return res.json()
+        except ConnectionError as ce:
+            self.logger.error(f'Database connection failed: {ce}')
```
Comment (asafMasa) [inline src/helper.py]: throw exception of unable to connect to DB - not null reference

File: src/exportImage.py [hunk 3]
```diff
@@ -56,4 +58,14 @@ def create_index(self, filename, fullPath):
         sql = f'CREATE unique INDEX tiles_index on {filename}(zoom_level, tile_column, tile_row)'
         data_source.ExecuteSQL(sql)
 
-
+    def runPreperation(self, taskId, filename):
+        status = self.__helper.get_status(taskId)
+        if status["status"] == Status.COMPLETED.value:
```
Comment (asafMasa) [inline src/exportImage.py]: add todo to change for better check (gdal timeout)

**Review by asafMasa — CHANGES_REQUESTED** [review]
🥇

File: src/exportImage.py (hunk 3 above)
Comment (cl-shaharshavit-rnd) [reply src/exportImage.py]: should remain regardless 
and the modification for the timeout is quite big and isn't related to this function

**Review by asafMasa — APPROVED** [review]

**Pull request merged**